x-pack/plugins/enterprise_search/public/applications/workplace_search/components/layout/nav.tsx

```diff
@@ -19,6 +19,7 @@ import {
   GROUPS_PATH,
   ORG_SETTINGS_PATH,
 } from '../../routes';
+import { useSettingsSubNav } from '../../views/settings/components/settings_sub_nav';
```

Comment:
**Optional:** I associate "use" with React hooks. What do you think about renaming this to `getSettingsSubNav`? Totally optional, this definitely works. If you agree feel free to do it in a different PR!

Reply:
++ I thought the same thing when I saw it.

Reply:
![homerhide](https://user-images.githubusercontent.com/1869731/122434079-65fdba00-cf5c-11eb-82c2-de97cbd4630c.gif)

Reply:
Also since we're discussing this here, I wanted to mention some unfortunately mildly annoying shenanigans with the `useXSubNav` hook pattern: 

1. Hooks can't ever be called conditionally. I tried to do things like this (for performance):
```js
items: useRouteMatch(SOURCES_PATH) ? useSourceSubNav() : []
```
But I got React errors because you're supposed to always call [hooks in a set order](https://reactjs.org/docs/hooks-rules.html). So I fell back to the hook itself returning early/undefined instead.

2. You can't use a hook within a hook/callback - I saw that the Observability folks [were using useMemo](https://github.com/elastic/kibana/pull/99380/files#diff-1f9338fabab6f332f6b4b11dbcec00f10b89097720c94b6087c53b186659c976R60) to memoize their side nav items/reduce rerenders, but unfortunately we can't nest a `useXSubNav` within a `useMemo`. It's probably not a _huge_ deal and our app is still super diffy, but it definitely feels like it would be nice to get around this.

The primary issue is, without it being a `useSubNav` fn, we can't use the Kea `useValues` hooks. And unfortunately we do need the actual `useValues` hook, we can't simply do `SourceLogic.values.id` - it crashes/errors on pages where `SourceLogic` isn't mounted.

There are a few ways I can think of to work around these limitations, but they're potentially complicated and require adding new infrastructure (e.g.: hook checks within the main nav itself but not the sub navs? useParams? an always-mounted logic file just for the nav which stores source/group IDs so we can avoid useValues?) and I tried to make this conversion as simple as possible for now.

I'm definitely interested in coming back to see if I can avoid hooks entirely for the sub navs - I might do so after the final WS pass if that's cool! Or someone else is totally welcome to look, but also we may be better off punting until 7.15 or until it becomes a major pain point.

Reply:
I agree that punting is fine for now.